utils/src/main/java/com/cloud/utils/ssh/SshHelper.java

```diff
@@ -58,6 +58,30 @@ public static void scpTo(String host, int port, String user, File pemKeyFile, St
         scpTo(host, port, user, pemKeyFile, password, remoteTargetDirectory, data, remoteFileName, fileMode, DEFAULT_CONNECT_TIMEOUT, DEFAULT_KEX_TIMEOUT);
     }
 
+    public static void scpFrom(String host, int port, String user, File permKeyFile, String localTargetDirectory, String remoteTargetFile) throws Exception {
```

Comment:
there is no issue with this PR but the SshHelper methods throwing a generic ```Exception``` is not a good pattern. At least it should be a SshHelper-specific wrapper :(

Reply:
I think it's going to be large effort to fix this pattern as part of this PR. Given the length of time this PR has been kept open and testing already done on this code perhaps we can clean up this in a separate PR?

Reply:
:) yes, but one of my big frustrations (pokemon tirany) is going to last, i'm afraid. I agree however it is not one for this PR